webtester-core/src/main/java/info/novatec/testit/webtester/mouse/SequenceWithFragment.java

```diff
@@ -0,0 +1,139 @@
+package info.novatec.testit.webtester.mouse;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+
+import info.novatec.testit.webtester.pagefragments.PageFragment;
+
+
+/**
+ * Provides the means of creating a sequence of mouse actions.
+ * This is mainly used for readable fluent API calls.
+ * <p>
+ * <b>Examples:</b>
+ * <pre>
+ * Mouse.sequence().moveTo(image).click();
+ * Mouse.sequence().moveTo(image).moveTo(otherImage).click();
+ * Mouse.sequence().click(image).doubleClick(otherImage);
+ * </pre>
+ *
+ * @see MouseDriver
+ * @see PageFragment
+ * @since 2.0
+ */
+@Getter(AccessLevel.PACKAGE)
+public class SequenceWithFragment {
```

Comment:
Hm, might work out in this case, I think.
In general, switching between return types of fluent API methods communicates context. Depending on context, different methods are valid to be called next (and offered by the IDE's auto-completion). Using inheritance here makes `SequenceWithFragment` a `Sequence`, and it will be impossible to define `Sequence` methods that should never be called in a `SequenceWithFragment` context. So we have less duplicated code at the cost of a potential loss of expressivity. Fluent APIs are all about expressivity, so this feels like a weird trade-off to make... :pensive: